be/src/vec/aggregate_functions/aggregate_function.h

```diff
@@ -287,8 +288,14 @@ class IAggregateFunctionHelper : public IAggregateFunction {
     void add_batch_single_place(size_t batch_size, AggregateDataPtr place, const IColumn** columns,
                                 Arena* arena) const override {
         const Derived* derived = assert_cast<const Derived*>(this);
-        for (size_t i = 0; i < batch_size; ++i) {
-            derived->add(place, columns, i, arena);
+
+        if constexpr (is_aggregate_function_multi_top<Derived>::value ||
+                      is_aggregate_function_multi_top_with_null_variadic_inline<Derived>::value) {
+            derived->add_range(place, columns, 0, batch_size, arena);
```

Comment:
add_range() calls add() in a loop just like the `else` branch. So is it necessary to add a branch for is_aggregate_function_multi_top?

```
    void add_range(AggregateDataPtr __restrict place, const IColumn** columns, ssize_t min,
                   ssize_t max, Arena* arena) const {
        for (ssize_t row_num = min; row_num < max; ++row_num) {
            add(place, columns, row_num, arena);
        }
    }
```

Reply:
Calling add directly on nullable types has poor performance. Using add_range uniformly can prevent multiple calls.